include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -557,11 +564,22 @@ struct alignment_configurator
 
             using scoring_scheme_policy_t = policy_scoring_scheme<config_t, alignment_scoring_scheme_t>;
 
-            // ----------------------------------------------------------------------------
-            // Configure alignment matrix policy
-            // ----------------------------------------------------------------------------
+            //----------------------------------------------------------------------------------------------------------
+            // Configure the alignment matrix policy.
+            //----------------------------------------------------------------------------------------------------------
 
-            using alignment_matrix_policy_t = policy_alignment_matrix<traits_t, score_matrix_single_column<score_t>>;
+            using score_matrix_t = score_matrix_single_column<score_t>;
+            using trace_matrix_t = trace_matrix_full<trace_directions>;
+
+            using alignment_matrix_t = std::conditional_t<traits_t::requires_trace_information,
+                                                          combined_score_and_trace_matrix<score_matrix_t,
+                                                                                          trace_matrix_t>,
+                                                          score_matrix_t>;
+            using alignment_matrix_policy_t = policy_alignment_matrix<traits_t, alignment_matrix_t>;
+
+            //----------------------------------------------------------------------------------------------------------
+            // Configure the final alignment algorithm.
+            //----------------------------------------------------------------------------------------------------------
 
             // ----------------------------------------------------------------------------
             // Configure alignment algorithm
```

Comment:
Now you have it double. Remove one of them

Reply:
Thanks for pointing this out. I remove it here right away